src/include/main/db_main.h

```diff
@@ -240,6 +241,8 @@ class DBMain {
       server_ = std::make_unique<network::TerrierServer>(common::ManagedPointer(provider_),
                                                          common::ManagedPointer(connection_handle_factory_),
                                                          thread_registry, port, connection_thread_count);
+      server_->RegisterProtocol(port, common::ManagedPointer(provider_), CONNECTION_THREAD_COUNT,
```

Comment:
Does this logic belong in the network layer? Or does it belong in TerrierServer? IMO it belongs in TerrierServer. Also, if you move it to TerrierServer you don't need to include settings.h.